src/sage/schemes/elliptic_curves/ell_finite_field.py

```diff
@@ -1818,15 +1839,21 @@ def twists(self):
 
         # Now j is not 0 or 1728, and we only have a quadratic twist
 
-        if K.characteristic() == 2: # find D with trace 1 for the additive twist
-            D = K.one() # will work if degree is odd
-            while D.trace() == 0:
+        if K.characteristic() == 2:
+            # find D with trace 1 for the additive twist
+            D = K.one()
+            while not D or D.trace() == 0:
                 D = K.random_element()
-        else: # find a nonsquare D
-            D = K.gen()
-            q2 = (K.cardinality()-1)//2
-            while not D or D**q2 == 1:
+        else:
+            # find a nonsquare D. note that computing a multiplicative generator is expensive
+            q2 = (K.cardinality() - 1) // 2
+            while True:
                 D = K.random_element()
```

Comment:
I don't understand the motivation behind this change: `K.gen()` doesn't compute a multiplicative generator.